File: core/src/core/utils/progress/tasks/LogLevel.ts
/**
 * Log level enumeration for task progress tracking.
 * Simple enum with three levels: WARNING, INFO, DEBUG.
 */
export enum LogLevel {
  WARNING = 'WARNING',
  INFO = 'INFO',
  DEBUG = 'DEBUG'
}

/**
 * Utility functions for LogLevel operations.
 */
export namespace LogLevel {
  /**
   * Parse string to LogLevel (case-insensitive).
   */
  export function fromString(level: string): LogLevel {
    const normalized = level.toUpperCase();
    switch (normalized) {
      case 'WARNING':
      case 'WARN':
        return LogLevel.WARNING;
      case 'INFO':
        return LogLevel.INFO;
      case 'DEBUG':
        return LogLevel.DEBUG;
      default:
        throw new Error(`Unknown log level: ${level}`);
    }
  }

  /**
   * Get numeric priority (higher = more important).
   */
  export function getPriority(level: LogLevel): number {
    switch (level) {
      case LogLevel.WARNING:
        return 3;
      case LogLevel.INFO:
        return 2;
      case LogLevel.DEBUG:
        return 1;
      default:
        return 0;
    }
  }

  /**
   * Check if level should be logged at given threshold.
   */
  export function shouldLog(level: LogLevel, threshold: LogLevel): boolean {
    return getPriority(level) >= getPriority(threshold);
  }

  /**
   * Get all log levels in order of priority.
   */
  export function getAllLevels(): LogLevel[] {
    return [LogLevel.DEBUG, LogLevel.INFO, LogLevel.WARNING];
  }

  /**
   * Get default log level.
   */
  export function getDefault(): LogLevel {
    return LogLevel.INFO;
  }
}

File: gds/src/core/utils/progress/tasks/log_level.rs
//! Log level enumeration for task progress tracking.

use std::fmt;

/// Log level for progress tracking messages.
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord, Default)]
pub enum LogLevel {
    Debug = 1,
    #[default]
    Info = 2,
    Warning = 3,
}

impl LogLevel {
    /// Get numeric priority (higher = more important).
    pub fn priority(self) -> u8 {
        self as u8
    }

    /// Check if this level should be logged at the given threshold.
    pub fn should_log(self, threshold: LogLevel) -> bool {
        self.priority() >= threshold.priority()
    }
}

impl fmt::Display for LogLevel {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        match self {
            LogLevel::Debug => write!(f, "DEBUG"),
            LogLevel::Info => write!(f, "INFO"),
            LogLevel::Warning => write!(f, "WARNING"),
        }
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_log_level_priority() {
        assert!(LogLevel::Warning.priority() > LogLevel::Info.priority());
        assert!(LogLevel::Info.priority() > LogLevel::Debug.priority());
    }

    #[test]
    fn test_should_log() {
        assert!(LogLevel::Warning.should_log(LogLevel::Debug));
        assert!(LogLevel::Info.should_log(LogLevel::Debug));
        assert!(!LogLevel::Debug.should_log(LogLevel::Info));
    }

    #[test]
    fn test_default() {
        assert_eq!(LogLevel::default(), LogLevel::Info);
    }
}
